connectivity.py

```diff
@@ -0,0 +1,100 @@
+from threading import RLock, Event
+
+from nio.properties import TimeDeltaProperty
+from nio.modules.scheduler import Job
+from nio.signal.status import BlockStatusSignal
+from nio.util.runner import RunnerStatus
+
+
+class PubSubConnectivity(object):
+    """ Adds connectivity awareness to pubsub blocks
+    """
+    timeout = TimeDeltaProperty(title='Connect Timeout',
+                                default={'seconds': 2},
+                                advanced=True)
+
+    def __init__(self):
+        super().__init__()
+        self._connected = None
+        self._connected_lock = RLock()
+        self._timeout_job = None
+        self._status_set = {RunnerStatus.warning: False,
+                            RunnerStatus.error: False}
+        self._connected_event = Event()
+
+    def conn_configure(self, is_connected):
+        """ Sets up instance for connectivity checks
+
+        Args:
+            is_connected (callable): function to invoke to establish initial
+                connectivity status
+        """
+        with self._connected_lock:
+            self._connected = is_connected()
+            self.logger.info("Starting in: '{}' state".format(
+                "connected" if self._connected else "disconnected"))
+
+        if not self._connected:
+            # per spec, hold the configure method hoping to get connected
+            if not self._connected_event.wait(self.timeout().total_seconds()):
```

Comment:
I was following the spec as detailed here: https://docs.google.com/document/d/1yFIy64fcZW_qHU0uW3SIUVwyPrdaQMS4KPLfzGjWrYA `nio Block Changes`

Reply:
The warning state part makes sense to me. I don't know why we would start a timeout job to wait another 2 seconds though before then going in to error state though. 

To me it seems like we should decide what state we want the block to be in if it can't connect to PK within the timeout. I think we agreed on warning state, which I am still ok with, but we can also discuss error state though if we want. But whatever that state is should get set after the timeout and then that's it. I don't think we need to run through the two states.

Reply:
We had discussed the block going into `error` state, which has its own consequences, let's talk about this when you get a chance